### PR TITLE
ENG-4008: Updating with symmetric decryption for lockbox

### PR DESCRIPTION
**Pull request by ankit819**
|JIRA    | [ENG-4008](https://eligible.atlassian.net/browse/ENG-4008)
|:-------|:-----------------------------------------------------------

### Description

Adding symmetric encryption for lockbox

### Checklists

**Development and testing:**
- [x] The code changed/moved as a part of this PR has been covered with tests
- [x] All tests related to the changed code pass in development

**Pre-release and code review:**
- [x] CI build is green
- [x] CodeClimate does not see any new issues
- [x] "Ready for review" label attached to PR and reviewers mentioned in a comment
- [x] Changes have been reviewed by at least one Senior Engineer
- [x] JIRA issue has a link to this pull request

### Related Pull Requests

List related PRs against other projects:

|project      | PR
|-------------|-----------------------
|models | [link](https://github.eligibleapi.com/EligibleAPI/models/pull/262)
|admin | [link](https://github.eligibleapi.com/EligibleAPI/admin/pull/471)
|gds | [link](https://github.eligibleapi.com/EligibleAPI/gds/pull/3032)

### CODE REVIEW

File: lib/eligible/lockbox.rb [hunk 1]
```diff
@@ -22,17 +23,20 @@ def self.delete_private_key!(params)
       params.delete(:private_key)
     end
 
-    def self.decrypt_data(data, private_key)
+    def self.decrypt_data(data, encrypted_key, private_key)
```
Comment (1989gaurav) [inline lib/eligible/lockbox.rb]: should we rename `encrypted_key` to `data_key`?

File: lib/eligible/lockbox.rb [hunk 2]
```diff
@@ -22,17 +23,20 @@ def self.delete_private_key!(params)
...
     def self.get_and_decrypt_from_lockbox(params, api_key = nil)
       private_key = extract_private_key(params)
       delete_private_key!(params)
-      req = get(params, api_key)
-      enc_data = req.to_hash[:encrypted_data]
-      decrypt_data(enc_data, private_key)
+      req = get(params, api_key).to_hash
+      enc_data = req[:encrypted_data]
+      enc_key = req[:data_key]
+      decrypt_data(enc_data, enc_key, private_key)
```
Comment (1989gaurav) [inline lib/eligible/lockbox.rb]: you can drop `enc_data` and `enc_key`.

**Comment by 1989gaurav** [comment]
LGTM

**Comment by ggeligible** [comment]
LGTM

**Comment by ggeligible** [comment]
LGTM

**Pull request merged**